go/app/infra.go

```diff
@@ -20,30 +30,159 @@ type Item struct {
 //go:generate go run go.uber.org/mock/mockgen -source=$GOFILE -package=${GOPACKAGE} -destination=./mock_$GOFILE
 type ItemRepository interface {
 	Insert(ctx context.Context, item *Item) error
+	List(ctx context.Context) ([]Item, error) //商品一覧を取得するためのメソッドを追加
+	Get(ctx context.Context, id string) (*Item, error)
 }
 
 // itemRepository is an implementation of ItemRepository
 type itemRepository struct {
 	// fileName is the path to the JSON file storing items.
 	fileName string
+	// filePath is the absolute path to the JSON file
+	filePath string
 }
 
 // NewItemRepository creates a new itemRepository.
 func NewItemRepository() ItemRepository {
-	return &itemRepository{fileName: "items.json"}
+	cwd, err := os.Getwd()
+	if err != nil {
+		// エラーハンドリングを簡略化、パニックを使用
+		panic(fmt.Errorf("failed to get working directory: %w", err))
+	}
+	fileName := "items.json"
+	return &itemRepository{
+		fileName: fileName,
+		filePath: filepath.Join(cwd, fileName),
+	}
+}
+
+// JSONファイルの内容をパースするための構造体
+type ItemsData struct {
+	Items []Item `json:"items"`
 }
 
 // Insert inserts an item into the repository.
 func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
-	// STEP 4-2: add an implementation to store an item
+
+  // STEP 4-2: add an implementation to store an item
+	// まずファイルの存在確認
+	if _, err := os.Stat(i.filePath); os.IsNotExist(err) {
+		return fmt.Errorf("items file not found: %w", err)
+	}
+
+	// JSONファイルを開く//読み書き
+	file, err := os.OpenFile(i.filePath, os.O_RDWR, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to open items file: %w", err)
+	}
+	defer file.Close()
+
+	// JSONの中身を読み取る
+	var data ItemsData
+	decoder := json.NewDecoder(file)
+	err = decoder.Decode(&data)
+	if err != nil && err != io.EOF {
+		return err
+	}
+	// IDを設定
+	item.ID = len(data.Items) + 1
+
+	// 新しいアイテムを追加
+	data.Items = append(data.Items, *item)
+
+	// ファイルを再度開いて内容をリセット
+	file.Seek(0, 0)
+	file.Truncate(0)
+
+	// 更新されたアイテムリストを書き込む
+	encoder := json.NewEncoder(file)
+	encoder.SetIndent("", "  ") // 見やすくする
+	err = encoder.Encode(data)
+	//err = json.NewEncoder(file).Encode(items)
+	if err != nil {
+		return err
+	}
+
 
 	return nil
 }
 
+// List returns all items from the repository.
+func (i *itemRepository) List(ctx context.Context) ([]Item, error) {
+	// JSONファイルを開く
+	file, err := os.OpenFile(i.filePath, os.O_RDONLY, 0644)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("items file not found: %w", err)
```

Comment:
Same as above, we should probably create the file here with a default value instead of returning the error